### PR TITLE
Add sonarqube scan to test workflow

### PR DESCRIPTION
**Pull request by violetbrina**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -23,12 +24,24 @@ jobs:
     - name: "Run unit tests"
       id: runtests
       run: |
-        coverage run -m unittest discover -p 'test*.py' -s '.'
+        coverage run -m pytest/ --junitxml=test-execution.xml
```
Comment (nevoodoo) [inline .github/workflows/test.yaml]: Maybe there's a typo here with the extra slash? Might be causing the error

**Review by nevoodoo — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yaml (hunk 1 above)
Comment (violetbrina) [reply .github/workflows/test.yaml]: Fixed!

**Review by nevoodoo — APPROVED** [review]

**Comment by jmarshall** [comment]
I realise I am late to the party, but: What is the reason for removing codecov?

**Comment by violetbrina** [comment]
Mostly that it's totally redundant. I didn't see much point in keeping it around if the coverage is in sonarqube anyway. Good question though. If you feel there is a codecov shaped whole in the repos let me know and we can always add it back. But it was only on 2 maybe 3 of our repos anyway.

**Pull request merged**